src/pfe/file-watcher/server/test/unit-test/tests/projectExtensions.module.test.ts

```diff
@@ -0,0 +1,328 @@
+    const appsodyExtensionTestArtifactPath1 = path.join(appsodyExtensionPath, ".sh-extension");
+    const appsodyExtensionTestArtifactPath2 = path.join(appsodyExtensionPath, "entrypoint.sh");
+
+    it("setProjectExtensionList: " + someRandomExtension + " & isProjectExtensionSupported: " + someRandomExtension, async () => {
+        projectExtensions.setProjectExtensionList(someRandomExtension);
+        const isExtensionSupported = projectExtensions.isProjectExtensionSupported(someRandomExtension);
+        expect(isExtensionSupported).to.equal(true);
+    });
+
+    it("removeProjectExtensionList: " + someRandomExtension + " & isProjectExtensionSupported: " + someRandomExtension, async () => {
+        projectExtensions.removeProjectExtensionList(someRandomExtension);
+        const isExtensionSupported = projectExtensions.isProjectExtensionSupported(someRandomExtension);
+        expect(isExtensionSupported).to.equal(false);
+    });
+
+    it("getAllProjectTypes" , async () => {
+        const projectTypes: string[] = projectExtensions.getAllProjectTypes();
+        expect(projectTypes.toString()).to.equal(defaultCodewindProjectTypes.toString());
+    });
+
+    describe("Combinational testing of getProjectTypes function", () => {
+
+        before("create test directories", async () => {
+            if (!(await existsAsync(libertyDummyProjectPath))) {
+                await mkdirAsync(libertyDummyProjectPath);
+                await copyAsync(originalLibertyPOM, testLibertyPOM);
+            }
+        });
+
+        after("remove test directories", async () => {
+            if ((await existsAsync(libertyDummyProjectPath))) {
+                await unlinkAsync(testLibertyPOM);
+                await rmdirAsync(libertyDummyProjectPath);
+            }
+        });
+
+        const combinations: any = {
+            "combo1": {
+                "location": undefined,
+                "result": defaultCodewindProjectTypes
+            },
+            "combo2": {
+                "location": libertyDummyProjectPath,
+                "result": ["liberty"]
+            }
+        };
+
+        for (const combo of Object.keys(combinations)) {
+            const location = combinations[combo]["location"];
+            const expectedResult = combinations[combo]["result"];
+
+            it(combo + " => getProjectTypes: " + location, async () => {
+                const projectTypes: string[] = await projectExtensions.getProjectTypes(location);
+                expect(projectTypes.toString()).to.equal(expectedResult.toString());
+            });
+        }
+    });
+
+    describe("Combinational testing of determineProjectType function", () => {
+
+        before("create test directories", async () => {
+            if (!(await existsAsync(libertyDummyProjectPath))) {
+                await mkdirAsync(libertyDummyProjectPath);
+                await copyAsync(originalLibertyPOM, testLibertyPOM);
```

Comment:
For any pom/dockerfiles that required for testing, can we clone it from git as a test pre-setup? Instead of saving those files under `resources` folder?
The dockerfiles would cause code scan complaint if we save them within our repo. So we need to clone anyways. better use the Pom from those cloned projects.